scripts/windows-download-cache-and-build-module-wheels.ps1

```diff
@@ -63,15 +76,26 @@ if ($env:ITKPYTHONPACKAGE_TAG) {
 }
 
 # Get other build dependencies
-Invoke-WebRequest -Uri "https://data.kitware.com/api/v1/file/5c0ad59d8d777f2179dd3e9c/download" -OutFile "doxygen-1.8.11.windows.bin.zip"
+if (-not (Test-Path doxygen-1.8.11.windows.bin.zip)) {
+  Invoke-WebRequest -Uri "https://data.kitware.com/api/v1/file/5c0ad59d8d777f2179dd3e9c/download" -OutFile "doxygen-1.8.11.windows.bin.zip"
+}
 7z x doxygen-1.8.11.windows.bin.zip -oC:\P\doxygen -aoa -r
-Invoke-WebRequest -Uri "https://data.kitware.com/api/v1/file/5bbf87ba8d777f06b91f27d6/download/grep-win.zip" -OutFile "grep-win.zip"
+if (-not (Test-Path grep-win.zip)) {
+  Invoke-WebRequest -Uri "https://data.kitware.com/api/v1/file/5bbf87ba8d777f06b91f27d6/download/grep-win.zip" -OutFile "grep-win.zip"
+}
 7z x grep-win.zip -oC:\P\grep -aoa -r
 $env:Path += ";C:\P\grep"
 
 # Build ITK module dependencies, if any
 $build_command = "& `"C:\Python$pythonVersion-x$pythonArch\python.exe`" `"C:\P\IPP\scripts\windows_build_module_wheels.py`" --no-cleanup --py-envs `"3$($args[0])-x64`""
-echo "Build command: $build_command"
+foreach ($arg in $args[1..$args.length]) {
+  if ($arg.substring(0,2) -eq "--") {
+    $build_command = "$build_command $($arg)"
```

Comment:
This looks weird to me. In your other comment you mention that quotes are mandatory for the `--` parameter, but here we don't preserve them.
What happens when sending `--` or `'"--'"` ?

Reply:
I will try to explain better. Let's assume you have a simple `test.ps1` powershell script:
```pwsh
foreach ($arg in $args[0..$args.length]) {
      echo $arg
}
```
 If you do `test.ps1 a b` or `test.ps1 a -- b`, you get
```
a
b
```
If you do `test.ps1 a "--" b`, you get
```
a
--
b
```
So I call e.g. `windows-download-cache-and-build-module-wheels.ps1 11 --exclude-libs nvcuda.dll "--" -DRTK_USE_CUDA:BOOL=ON` to later on detect the -- argument and pass it to `windows_build_module_wheels.py` without quotes.
Does that maker more sense? It's a bit dirty but... any better alternative to suggest?

Reply:
Thanks a lot for the explanation Simon. Do you think the following approach would work?:
Escape the -- argument using ``` `-- ``` when passing it to the ps1 script. The ps1 script should then receive `--` without quotes. Then it can be forwarded to the python script with quotes.

Reply:
No since if you do `test.ps1 a b` or `test.ps1 a -- b`, you always get
```
a
b
```
i.e. no trace of -- (as far as my poor powershell knowledge goes).

Reply:
In the Windows Powershell console, I do receive -- if I pass it using ```  `--  ``` (note the `` ` `` to escape).

```
 PS C:\Users\Lucas\Projects> ./test.ps1 a `-- b
a
--
b
```

But maybe that does not help with your issue as you still need to forward it to the python script without quote?

Reply:
Sorry, I missed the escape quote! I guess this would also work indeed but I don't think I'd need to change the proposed code (all arguments are passed between quotes except the ones starting with `--` which should not use quotes to be processed). Do you think I should modify the [documentation](https://github.com/InsightSoftwareConsortium/ITKPythonPackage/pull/248/files#diff-792f0189775779f5115d0f33b9658339c41e690bfd50428e44b1164496c40fd7R19) to mention this alternative option (to `"--"`)?

Reply:
No worries, then I agree there is no need to update the proposed code if the python script must receive `--` without quotes to process it. So no need to update the documention I think.